### PR TITLE
[polish] improve error message for missing devbox-json

### PR DESCRIPTION
**Pull request by savil**
## Summary

The old message was not very nice.

```
> devbox shell empty-projects/no-devbox-json
Error: open /Users/savil/code/jetpack/devbox/testdata/empty-projects/no-devbox-json/devbox.json: no such file or directory
```
or

```
> devbox shell
Error: open devbox.json: no such file or directory
````
## How was it tested?

inside the project directory:
```
➜ devbox shell
Error: No devbox.json found in this directory. Did you run `devbox init` yet?
```

when using a path, print the relative dir:
```
➜ devbox shell empty-projects/no-devbox-json
Error: No devbox.json found in empty-projects/no-devbox-json. Did you run `devbox init` yet?
```

### CODE REVIEW

**Comment by savil** [comment]
Current dependencies on/for this PR:
* main
  * **PR #174** <a href="https://app.graphite.dev/github/pr/jetpack-io/devbox/174" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jetpack-io/devbox/174?utm_source=stack-comment).

File: devbox.go [hunk 1]
```diff
@@ -20,6 +21,26 @@ import (
...
+		wd, err := os.Getwd()
+		// if an error occurs, then just use `dir`
+		if err == nil {
+			relDir, err := filepath.Rel(wd, dir)
+			if err == nil {
+				dir = relDir
+			}
+		}
+	}
+	return errors.Errorf("No devbox.json found in %s. Did you run `devbox init` yet?", dir)
```
Comment (mikeland86) [inline devbox.go:41]: Use `usererr`

File: devbox.go [hunk 2]
```diff
@@ -20,6 +21,26 @@ import (
 // configFilename is name of the JSON file that defines a devbox environment.
 const configFilename = "devbox.json"
 
+var errNoDevboxJSON = func(dir string) error {
```
Comment (mikeland86) [inline devbox.go:24]: feels a bit nonstandard to declare function in this way?

**Review by mikeland86 — APPROVED** [review]

File: devbox.go (hunk 2 above)
Comment (savil) [reply devbox.go:24]: oh yeah, its highly non-standard, not a bit ;)

I'll edit it to be a regular function called `resolveMissingDevboxJsonError`

File: devbox.go (hunk 1 above)
Comment (savil) [reply devbox.go:41]: ah TIL - will change.

**Pull request merged**